contracts/Synthswap.sol

```diff
@@ -24,25 +28,61 @@ contract SynthSwap is ISynthSwap {
         bytes calldata payload,
         bytes32 destinationSynthCurrencyKey
     ) external payable override returns (uint) {
+
         // Make sure to set destReceiver to this contract
         (bool success, bytes memory returnData) = aggregationRouterV3.delegatecall(payload);
         require(success, _getRevertMsg(returnData));
         require(false, "stop");
+
+        // Approve the Synthetix router to spend sUSD.
         uint sUSDBalance = sUSD.balanceOf(address(this));
         sUSD.approve(address(Synthetix), sUSDBalance);
 
+        // Swap sUSD with destination synth by providing both the source and destination currency keys. 
         uint amountReceived = Synthetix.exchangeWithTrackingForInitiator(
-            'sUSD', // hardcode source currency key
-            sUSDBalance, //source amount
-            destinationSynthCurrencyKey,
-            volumeRewards, 
-            'KWENTA' //tracking code
+            SUSD_CURRENCY_KEY, // hardcode source currency key
+            sUSDBalance, // source amount
+            destinationSynthCurrencyKey, // destination currency key
+            volumeRewards, // volume rewards address
+            'KWENTA' // tracking code
         );
         
         emit SwapInto(msg.sender, amountReceived);
         return amountReceived;
     }
 
+    function swapOutOf(
+        bytes calldata payload,
+        address inputSynth,
+        bytes32 inputSynthCurrencyKey,
+        address destinationToken
+    ) external payable override returns (uint) {
+
+        // Approve the Synthetix router to spend `inputSynth`.
+        IERC20 InputERC20 = IERC20(inputSynth);
+        uint synthBalance = InputERC20.balanceOf(address(this));
```

Comment:
We should be careful because there are a couple implications to consider here:

1. If the user deposits less inputSynth, subsequently less outputSynth than the amount specified in the 1inch API call -- does the low-level call to the 1inch swap revert?
2. If someone tries to grief attack by increasing the balanceOf this contract, how does that affect the current execution of this function and how does it affect the future execution of this function (or even `swapInto()`)